File: personal-git-repo-alimkanova-t-auca-2022-main/extra/translation/URI1558/URI1558.java
import java.util.Scanner;

public class URI1558 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        while (input.hasNextInt()) {
            boolean isRight = false;
            int num = input.nextInt();
            int maxOfNum = (int) Math.round(Math.sqrt(Math.abs(num)));
            for (int i = 0; i <= maxOfNum; ++i) {
                for (int j = 0; j <= maxOfNum; ++j) {
                    if (Math.pow(i, 2) + Math.pow(j, 2) == num) {
                        isRight = true;
                        break;
                    }
                }
            }
            if (isRight) {
                System.out.println("YES");
            } else {
                System.out.println("NO");
            }
        }
    }
}

File: cpp-projects-2022/extra/translation/URI1558/URI1558.cpp
//
// Created by User on 17.09.2023.
//
#include "iostream"
#include "cmath"

using namespace std;

int main() {
    int num;
    while (scanf("%d", &num) != EOF) {
        bool isRight = false;
        int maxOfNum = (int) round(sqrt(abs(num)));
        for (int i = 0; i <= maxOfNum; ++i) {
            for (int j = 0; j <= maxOfNum; ++j) {
                if (pow(i, 2) + pow(j, 2) == num) {
                    isRight = true;
                    break;
                }
            }
            if (isRight) {
                break;
            }
        }
        if (isRight) {
            cout << "YES" << endl;
        } else {
            cout << "NO" << endl;
        }
    }
    return 0;
}